plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -769,7 +771,14 @@ protected synchronized void initializeStorage() throws IOException {
                             "This is expected to happen when using the " + durabilitySetting + " durability setting and Jenkins is not shut down cleanly. " +
                             "Consider investigating to understand if Jenkins was not shut down cleanly or switching to the MAX_SURVIVABILITY durability setting which should prevent this issue in most cases.");
                 } else {
-                    throw new AbortException("Cannot resume build because FlowNode " + entry.getValue() + " for FlowHead " + entry.getKey() + " could not be loaded.");
+                    var sd = getStorageDir().toPath();
+                    List<Path> files = null;
+                    try (var walk = Files.walk(sd)) {
+                        files = walk.filter(Files::isRegularFile).map(sd::relativize).sorted().toList();
```

Comment:
Would it be interesting to also report `Files.getLastModifiedTime`, in case the problem you are investigating is somehow related to NFS caching?